build/package/Dockerfile

```diff
@@ -1,14 +1,15 @@
-FROM golang:alpine as builder
-RUN mkdir /build 
-ADD . /build/
-COPY go.mod /go/src/github.com/flostadler/name-generator/cmd/name-generator/go.mod
-COPY go.sum /go/src/github.com/flostadler/name-generator/cmd/name-generator/go.sum
-RUN apk add git
-WORKDIR /build
-RUN go mod download
-RUN go build -o main github.com/flostadler/name-generator/cmd/name-generator
-FROM alpine
-EXPOSE 80
-COPY --from=builder /build/main /app/
+FROM golang:alpine3.16
+
+COPY web/build /app/
+COPY go.mod /app/go.mod
+COPY go.sum /app/go.sum
+
 WORKDIR /app
-CMD ["./main"]
+
+RUN apk add --no-cache git
+RUN go install github.com/shurcooL/goexec@latest
+RUN go get github.com/shurcooL/go-goon
+
+EXPOSE 80
+
+CMD ["goexec", "http.ListenAndServe(`:80`, http.FileServer(http.Dir(`.`)))"]
```

Comment:
What's the reason for adding goexec and go-goon?
go-goon is deprecated afaik.

Why can't we just have the mux http APIs as part of a regular go executable?

Or is this for starting a web server to serve the web site locally?
If it's for local development I'd say we should try and keep both the APIs and the local development setup.

I think for the local development using nginx as the container image might be easiest

Reply:
@flostadler thanks for the review! I'm pretty new at Go development so in this case I've followed one of the recommended approaches on containerizing Go+WASM application. Guidelines might not have been 100% up-to-date though since you mentioned that go-goon is deprecated.
Now, I'm not quite sure why you need to build a container to only have APIs? I thought that the purpose of introducing a container in this case is to be able to run and host this application wherever you want instead of using publicly available version for instance... Therefore I updated Dockerfile with a thought to make it possible to run Sprint Name Generator application as a whole, in a container. I tried to make the original container work but I didn't manage it, maybe due to the fact that I didn't have the whole picture on why you would want to only make the APIs available in a container and which APIs would that be?

I can update the Dockerfile by removing the outdated components so that an application could be run as a container but should the API version also be available you think?

Also, please check PR description - I think that you will need to make a few adjustments to make the build work once all changes in this PR are OK.

Reply:
@flostadler , today I digged more into how Golang and WASM hang together and of course it's perfectly simple and straightforward to use NGINX as a web server in this case! Thanks a lot for challenging me on this, learned a lot of valuable stuff today! 😸 I've now tested and updated the Dockerfile to support running Sprint Name Generator in a container. Please let me know if anything else needs to be updated.